Content.Server/TapeRecorder/TapeRecorderSystem.cs

```diff
@@ -0,0 +1,340 @@
+using System.Linq;
+using Content.Server.Chat;
+using Content.Server.Chat.Systems;
+using Content.Server.Popups;
+using Content.Shared.Chat;
+using Content.Shared.Containers.ItemSlots;
+using Content.Shared.Examine;
+using Content.Shared.Interaction.Events;
+using Content.Shared.Verbs;
+using Content.Shared.TapeRecorder;
+using Robust.Shared.Player;
+using Robust.Server.Containers;
+using Robust.Shared.Containers;
+
+namespace Content.Server.TapeRecorder
+{
+    /// <summary>
+    /// System for handling tape recorders
+    /// </summary>
+    public sealed class TapeRecorderSystem : EntitySystem
+    {
+        [Dependency] private readonly ChatSystem _chat = default!;
+        [Dependency] private readonly PopupSystem _popupSystem = default!;
+        [Dependency] private readonly SharedAudioSystem _audioSystem = default!;
+        [Dependency] private readonly ContainerSystem _containerSystem = default!;
+        [Dependency] private readonly SharedAppearanceSystem _appearanceSystem = default!;
+
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<TapeRecorderComponent, ChatMessageHeardNearbyEvent>(OnChatMessageHeard);
+            SubscribeLocalEvent<TapeRecorderComponent, UseInHandEvent>(OnUseInHand);
+            SubscribeLocalEvent<TapeRecorderComponent, GetVerbsEvent<AlternativeVerb>>(OnGetAltVerbs);
+            SubscribeLocalEvent<TapeRecorderComponent, ExaminedEvent>(OnExamined);
+            SubscribeLocalEvent<TapeRecorderComponent, ItemSlotChangedEvent>(OnItemSlotChanged);
+        }
+
+        public override void Update(float frameTime)
+        {
+            base.Update(frameTime);
+
+            foreach (var tapeRecorder in EntityManager.EntityQuery<TapeRecorderComponent>())
+            {
+                tapeRecorder.AccumulatedTime += frameTime;
+
+                if (tapeRecorder.InsertedTape == null) //just dont bother without a tape inside
+                    return;
```

Comment:
```suggestion
                    continue;
```

Also seems weird to just keep accumulating time and then specifying a start time? Seems like it would lead to imprecision issues on long-running servers when you could just accumulate only active tape recorders?